aspnetcore/host-and-deploy/docker/visual-studio-tools-for-docker.md

```diff
@@ -46,53 +48,119 @@ If the target framework is .NET Core, the **OS** drop-down allows for the select
+
+### Service Fabric
+
+In addition to the base [Prerequisites](#prerequisites), the [Service Fabric](/azure/service-fabric/) orchestration solution demands the following prerequisites:
+
+* [Microsoft Azure Service Fabric SDK](https://www.microsoft.com/web/handlers/webpi.ashx?command=getinstallerredirect&appid=MicrosoftAzure-ServiceFabric-CoreSDK) version 2.6 or later
+* Visual Studio 2017's **Azure Development** workload
+
+Service Fabric doesn't support running Linux containers in the local development cluster on Windows. If the project is already using a Linux container, Visual Studio prompts to switch to Windows containers.
+
+The Visual Studio Tools for Docker do the following tasks:
+
+* Adds a *&lt;project_name&gt;Application* **Service Fabric Application** project to the solution.
+* Adds a *Dockerfile* and a *.dockerignore* file to the ASP.NET Core project. If a *Dockerfile* already exists in the ASP.NET Core project, it's renamed to *Dockerfile.original*. A new *Dockerfile*, similar to the following, is created:
+
+    [!code-dockerfile[](visual-studio-tools-for-docker/samples/2.1/HelloDockerTools/Dockerfile)]
+
+* Adds an `<IsServiceFabricServiceProject>` element to the ASP.NET Core project's *.csproj* file:
+
+    [!code-xml[](visual-studio-tools-for-docker/samples/2.1/HelloDockerTools/HelloDockerTools.csproj?name=snippet_IsServiceFabricServiceProject)]
+
+* Adds a *PackageRoot* folder to the ASP.NET Core project. The folder includes the service manifest and settings for the new service.
+
+For more information, see [Deploy a .NET app in a Windows container to Azure Service Fabric](/azure/service-fabric/service-fabric-host-app-in-a-container).
 
 ## Debug
 
 Select **Docker** from the debug drop-down in the toolbar, and start debugging the app. The **Docker** view of the **Output** window shows the following actions taking place:
 
+::: moniker range=">= aspnetcore-2.1"
+
+* The *2.1-aspnetcore-runtime* tag of the *microsoft/dotnet* runtime image is acquired (if not already in the cache). The image installs the ASP.NET Core and .NET Core runtimes and associated libraries. It's optimized for running ASP.NET Core apps in production.
+* The *ASPNETCORE_ENVIRONMENT* environment variable is set to `Development` within the container.
+* Two dynamically assigned ports are exposed: one for HTTP and one for HTTPS. The port assigned to localhost can be queried with the `docker ps` command.
+* The app is copied to the container.
+* The default browser is launched with the debugger attached to the container using the dynamically assigned port.
+
+The resulting Docker image of the app is tagged as *dev*. The image is based on the *2.1-aspnetcore-runtime* tag of the *microsoft/dotnet* base image. Run the `docker images` command in the **Package Manager Console** (PMC) window. The images on the machine are displayed:
+
+```console
+REPOSITORY          TAG                      IMAGE ID            CREATED             SIZE
```

Comment:
Three spots (here and below) ...

Do these blow the 80-85 char width we discussed previously? I see a small opportunity to narrow the columns here and below ... although ... it isn't by much. I agree tho with use of a code block ... and if it runs wide ... oh, well ... it is what it is. lol

Reply:
I'll narrow them a bit, but there's only so much that can be done with console output.